### PR TITLE
Refactor lineplot object

### PR DESCRIPTION
**Pull request by starcalibre**
The first of the promised series of PR.. this one is an example of how I've been re-writing the plot objects. Once this one is merged, I can send PRs for the rest of the plots then start adding some new features to the UI (excite!).

I've mostly just broken up lots of the larger functions into smaller private methods. Note: JS has no true private methods, but it's convention to use the ``_`` suffix for any method to be treated as private. (same as Python, right?)

### CODE REVIEW

File: public/js/Lineplot.js [hunk 1]
```diff
@@ -209,56 +216,90 @@ Lineplot.prototype.updateActiveLine = function(feature) {
...
+
+/**
+ * updateActiveLine: Redraw the line overlaying selected feature in the plot.
  *
  * @this {Lineplot}
+ * @private
  */
-Lineplot.prototype.destroy = function() {
-    d3.select(this.element + ' > svg').remove();
+Lineplot.prototype._updateActiveLine = function() {
```
Comment (jni) [inline public/js/Lineplot.js]: "Lineplot" is a very generic name, whereas this function is for a very specific line in a very specific plot. So there's some API tweaking to be done here...

Comment (starcalibre) [inline public/js/Lineplot.js]: ``FeatureVectorLinepot`` and and the equally creative ``FeatureDistributionHistogram`` for the histogram ?

Comment (jni) [inline public/js/Lineplot.js]: Yes these two names sound appropriate. =) Well, `LinePLOT`, not `LinePOT`. =P

File: public/js/script.js [hunk 2]
```diff
@@ -117,7 +117,7 @@ function mountPlots(screenData, sampleData, featureNames) {
 
     var neighbourImages = new NeighbourImages();
     var histogram = new Histogram(screenData._id, featureNames, '#histplot');
-    var lineplot = new Lineplot(screenData._id, '#lineplot');
+    var lineplot = new Lineplot('#lineplot');
```
Comment (jni) [inline public/js/script.js]: Same here re:naming. Make sure to name variables as meaningfully as possible.

File: public/js/Lineplot.js (hunk 1 above)
Comment (jni) [inline public/js/Lineplot.js]: Oh and again, in this context, "updateActiveFeatureVline", for example

File: public/js/Lineplot.js [hunk 3]
```diff
@@ -209,56 +216,90 @@ Lineplot.prototype.updateActiveLine = function(feature) {
...
+    var xCoord = this.xScale(this.activeFeature);
+    this.svg.selectAll('.selectedLine').remove();
+    this.svg.append('line')
+        .attr('x1', xCoord)
+        .attr('y1', 0)
+        .attr('x2', xCoord)
+        .attr('y2', this.height)
+        .attr('stroke', 'red')
+        .attr('stroke-width', 0.75)
```
Comment (jni) [inline public/js/Lineplot.js]: magic numbers are... baaaaaad.

File: public/js/Lineplot.js [hunk 4]
```diff
@@ -209,56 +216,90 @@ Lineplot.prototype.updateActiveLine = function(feature) {
...
+        .attr('stroke-dasharray', '5, 5')
```
Comment (jni) [inline public/js/Lineplot.js]: ditto. What's with the string type, btw?

File: public/js/Lineplot.js (hunk 3 above)
Comment (starcalibre) [inline public/js/Lineplot.js]: I should probably define these styles in the CSS anyway!

File: public/js/Lineplot.js [hunk 5]
```diff
@@ -39,24 +38,25 @@ function Lineplot(screen, element) {
...
     var aspectHeight = config.aspectRatio.height;
     this.fullWidth = $(this.element).width();
-    this.fullHeight = Math.round(this.fullWidth * (aspectHeight/aspectWidth));
+    this.fullHeight = Math.round(this.fullWidth *
+        (aspectHeight / aspectWidth));
 
-    this.width = this.fullWidth - this.margin.left - this.margin.right;
-    this.height = this.fullHeight - this.margin.top - this.margin.bottom;
+    this.width = this.fullWidth - this.margin.left -
+        this.margin.right;
```
Comment (jni) [inline public/js/Lineplot.js]: Why is this on two lines? It's less than 80c...

File: public/js/Lineplot.js [hunk 6]
```diff
@@ -39,24 +38,25 @@ function Lineplot(screen, element) {
     var aspectWidth = config.aspectRatio.width;
     var aspectHeight = config.aspectRatio.height;
     this.fullWidth = $(this.element).width();
-    this.fullHeight = Math.round(this.fullWidth * (aspectHeight/aspectWidth));
+    this.fullHeight = Math.round(this.fullWidth *
+        (aspectHeight / aspectWidth));
```
Comment (jni) [inline public/js/Lineplot.js]: This line break seems unnecessary as well

File: public/js/Lineplot.js (hunk 4 above)
Comment (starcalibre) [inline public/js/Lineplot.js]: It's an SVG attribute, they always take strings as their type. Numbers just end up getting parsed as strings anyway :)

eg.
```
<svg height="210" width="500">
  <line x1="0" y1="0" x2="200" y2="200" style="stroke:rgb(255,0,0);stroke-width:2" />
</svg>
```

File: public/js/Lineplot.js [hunk 7]
```diff
@@ -65,139 +65,146 @@ Lineplot.prototype.getSampleData = function(sampleId) {
...
         .call(yAxis);
+};
+
+/**
+ * drawLine: Draw the line for the lineplot.
+ *
+ * @private
+ */
+Lineplot.prototype._drawLine = function() {
+    var linePoints = _.zip(_.range(1, this.featureVector.length + 1),
```
Comment (jni) [inline public/js/Lineplot.js]: Note here why you're using range(1, n+1)

File: public/js/Lineplot.js (hunk 6 above)
Comment (starcalibre) [inline public/js/Lineplot.js]: 81 characters without the break.. :C

**Comment by jni** [comment]
You are in like Flynn! =P

**Pull request merged**